src/bam/call_consensus_reads/pipeline.rs

```diff
@@ -65,7 +70,13 @@ impl CallConsensusRead {
                         Some(record_pair) => {
                             //For right record save end position and duplicate group ID
                             group_end_idx
-                                .entry(record.cigar_cached().unwrap().end_pos()? - 1)
+                                .entry(
+                                    record.cigar_cached().unwrap().end_pos().context(
+                                        errors::BamCigarError {
+                                            record: record.clone(),
```

Comment:
This clone will happen always, causing unnecessary work if no error occurs. Can context take a closure or something like that?

Reply:
That has now been changed to return the cigar string itself instead the record.